### PR TITLE
VReplication: End to End tests with a few related bug fixes

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**

### CODE REVIEW

File: config/init_db.sql [hunk 1]
```diff
@@ -49,6 +49,9 @@ CREATE TABLE IF NOT EXISTS _vt.shard_metadata (
 CREATE USER 'vt_dba'@'localhost';
 GRANT ALL ON *.* TO 'vt_dba'@'localhost';
 GRANT GRANT OPTION ON *.* TO 'vt_dba'@'localhost';
+CREATE USER 'vt_dba'@'127.0.0.1';
```
Comment (sougou) [inline config/init_db.sql]: Is this change required for the tests to work?

**Comment by sougou** [comment]
Can you also verify that all your commits are signed?

File: config/init_db.sql [hunk 2]
```diff
@@ -49,6 +49,9 @@ CREATE TABLE IF NOT EXISTS _vt.shard_metadata (
 CREATE USER 'vt_dba'@'localhost';
 GRANT ALL ON *.* TO 'vt_dba'@'localhost';
 GRANT GRANT OPTION ON *.* TO 'vt_dba'@'localhost';
+#CREATE USER 'vt_dba'@'127.0.0.1';
```
Comment (sougou) [inline config/init_db.sql]: Many people use this config in production. So, it's better to delete the commented out code.

**Comment by sougou** [comment]
Also, looks like there's a new data race in unit_race test.

File: config/init_db.sql (hunk 2 above)
Comment (rohit-nayak-ps) [reply config/init_db.sql]: Sorry, oversight. Meant to remove the lines after testing. Fixed.

**Comment by rohit-nayak-ps** [comment]
> Also, looks like there's a new data race in unit_race test.
Yes, wrongly scoped participants variable. Fixed and test passes now.
